app/actors/ProxyConnection.java

```diff
@@ -0,0 +1,231 @@
+            _destinationClient.connect();
+        } else if (message instanceof DestinationConnected) {
+            // If we have both connections tie them together and flush buffers
+            if (_originatorIn != null && _originatorOut != null) {
+                establishProxy();
+            }
+        } else if (message instanceof OriginatorMessage) {
+            final OriginatorMessage originatorMessage = (OriginatorMessage) message;
+            // No need to lock _isProxied since Akka guarantees this is single threaded.
+            if (_isProxied.get()) {
+                _destinationClient.send(originatorMessage.getMessage());
+            } else {
+                _originatorMessageQueue.add(originatorMessage.getMessage());
+            }
+        }
+    }
+
+    private void establishProxy() {
+        LOGGER.info()
+                .setMessage("Established proxy connection")
+                .addData("actor", self())
+                .addData("destination", _destinationClient.getURI())
+                .log();
+
+        // Proxy connection established
+        if (ProxyConnection.this._isProxied.getAndSet(true)) {
+            throw new IllegalStateException("Connection is already proxied");
+        }
+        _isProxied.set(true);
+
+        // Send any buffered messages from the originator to the destination
+        while (!_originatorMessageQueue.isEmpty()) {
+            _destinationClient.send(_originatorMessageQueue.remove());
+        }
+
+        // Send any buffered messages from the destination to the orignator
+        _destinationClient.establishProxy();
+    }
+
+    private final MetricsFactory _metricsFactory;
+    private final AtomicBoolean _isProxied = new AtomicBoolean(false);
+    private final Queue<String> _originatorMessageQueue = new LinkedList<>();
+    private WebSocket.In<String> _originatorIn;
+    private WebSocket.Out<String> _originatorOut;
+    private ProxyWebSocketClient _destinationClient;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProxyConnection.class);
+
+    private static final class DestinationConnected {}
+
+    private static final class OriginatorMessage {
+        OriginatorMessage(final String message) {
+            _message = message;
+        }
+
+        public String getMessage() {
+            return _message;
+        }
+
+        private final String _message;
+    }
+
+    private final class ProxyWebSocketClient extends WebSocketClient {
+
+        ProxyWebSocketClient(final URI uri) {
+            super(uri);
+        }
+
+        @Override
+        public void onOpen(final ServerHandshake handshakedata) {
+            ProxyConnection.this.getSelf().tell(new DestinationConnected(), ProxyConnection.this.getSelf());
+        }
+
+        @Override
+        public void onMessage(final String message) {
+            // Unsure what the underlying threading model in WebSocketClient is; so play it safe
+            synchronized (_isProxied) {
```

Comment:
Definitely not the way to go here.  You should be firing messages to the ProxyConnection actor for it to write to the originator.